### PR TITLE
LAU-754 Token generation before application run

### PR DESCRIPTION
**Pull request by priyankaVerma21**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/LAU-754

### Change description ###
Generating token before application start

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/idam/ApplicationExecutor.java [hunk 1]
```diff
@@ -18,12 +19,14 @@ public class ApplicationExecutor implements ApplicationRunner {
...
 
+    private final SecurityUtil securityUtil;
+
     @Override
     @SuppressWarnings("PMD.AvoidThrowingRawExceptionTypes")
     public void run(ApplicationArguments args) {
-
         if (isServiceEnabled) {
             log.info("Starting the Idam-Disposer job...");
+            securityUtil.generateTokens();
```
Comment (MartinYSpasov) [inline src/main/java/uk/gov/hmcts/reform/idam/ApplicationExecutor.java:29]: nothing stopping you to generate the tokens before the _while_ loop in the run() method.

File: src/main/java/uk/gov/hmcts/reform/idam/service/aop/MethodRetryAspect.java [hunk 2]
```diff
@@ -36,6 +36,7 @@ public Object retry(ProceedingJoinPoint joinPoint) throws Throwable {
                 successful = true;
             } catch (FeignException fe) {
                 if (retryAttempts >= 0 && fe.status() == HttpStatus.SC_UNAUTHORIZED) {
+                    log.info("Security Util generate tokens are going to be called from retry");
```
Comment (MartinYSpasov) [inline src/main/java/uk/gov/hmcts/reform/idam/service/aop/MethodRetryAspect.java:39]: This log looks like debug log that might not necessarily help to identify a potential issue. Please try to log more meaningful information such as - which endpoint has thrown the exception for example.

**Review by MartinYSpasov — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/idam/ApplicationExecutor.java (hunk 1 above)
Comment (priyankaVerma21) [reply src/main/java/uk/gov/hmcts/reform/idam/ApplicationExecutor.java:29]: There is no while loop and if service is not enabled then why we want to generate the token

Comment (MartinYSpasov) [reply src/main/java/uk/gov/hmcts/reform/idam/ApplicationExecutor.java:29]: I meant in IdamUserDisposerService.run() method.

**Review by MartinYSpasov — APPROVED** [review]

**Review by MartinYSpasov — APPROVED** [review]

**Pull request merged**